File: csharp/calculator-conundrum/CalculatorConundrum.cs
using System;

public static class SimpleCalculator
{
    public static string Calculate(int operand1, int operand2, string operation)
    {
        if (operation == null)
        {
            throw new ArgumentNullException();
        }

        int outcome;
        if (operation.Equals("+"))
        {
            outcome = operand1 + operand2;
        }
        else if (operation.Equals("*"))
        {
            outcome = operand1 * operand2;
        }
        else if (operation.Equals("/"))
        {
            if (operand2 == 0)
            {
                return "Division by zero is not allowed.";
            }

            outcome = operand1 / operand2;
        }
        else if (operation.Equals(""))
        {
            throw new ArgumentException();
        }
        else
        {
            throw new ArgumentOutOfRangeException();
        }

        return $"{operand1} {operation} {operand2} = {outcome}";
    }
}

File: java/calculator-conundrum/src/main/java/CalculatorConundrum.java
class CalculatorConundrum {
  public String calculate(int operand1, int operand2, String operation) {
    if (operation == null) {
      throw new IllegalArgumentException("Operation cannot be null");
    }
    if (operation.isEmpty()) {
      throw new IllegalArgumentException("Operation cannot be empty");
    }

    if (operation.equals("+")) {
      return "%d + %d = %d".formatted(operand1, operand2, operand1 + operand2);
    }
    if (operation.equals("*")) {
      return "%d * %d = %d".formatted(operand1, operand2, operand1 * operand2);
    }
    if (operation.equals("/")) {
      try {
        return "%d / %d = %d".formatted(operand1, operand2, operand1 / operand2);
      } catch (ArithmeticException e) {
        throw new IllegalOperationException("Division by zero is not allowed", e);
      }
    }

    throw new IllegalOperationException("Operation '%s' does not exist".formatted(operation));
  }
}
